File: src/Game/Combat/PvE/PvETargetManager.ts
import { PvETarget } from './PvETarget';
import { Character } from '../../Character/Character';
import { CharacterType } from '../../Character/CharacterType';
import { Equipment } from '../../Equipment/Equipment';
import { EquipmentType } from '../../Equipment/EquipmentType';
import { EquipmentSlot } from '../../Equipment/EquipmentSlot';
import { Stats } from '../../Stats/Stats';

export class PvETargetManager {
    public static getTarget(target: PvETarget): Character {
        switch (target) {
            case PvETarget.WoodenDummy:
                return new Character(
                    CharacterType.WoodenDummy,
                    new Stats(10, 0, 0, 0, 0, 0),
                    new Equipment(EquipmentType.WoodenDummy, EquipmentSlot.None)
                );
            case PvETarget.Chicken:
                return new Character(
                    CharacterType.Chicken,
                    new Stats(15, 2, 1, 0, 0, 0),
                    new Equipment(EquipmentType.None, EquipmentSlot.None)
                );
            case PvETarget.Dogo:
                return new Character(
                    CharacterType.Dogo,
                    new Stats(20, 3, 2, 0, 0, 0),
                    new Equipment(EquipmentType.None, EquipmentSlot.None)
                );
            case PvETarget.DaphneBlake:
                return new Character(
                    CharacterType.DaphneBlake,
                    new Stats(25, 4, 3, 1, 1, 1),
                    new Equipment(EquipmentType.None, EquipmentSlot.None)
                );
            case PvETarget.Thug:
                return new Character(
                    CharacterType.Thug,
                    new Stats(30, 5, 4, 2, 2, 2),
                    new Equipment(EquipmentType.Club, EquipmentSlot.Weapon)
                );
            case PvETarget.Sellsword:
                return new Character(
                    CharacterType.Sellsword,
                    new Stats(35, 6, 5, 3, 3, 3),
                    new Equipment(EquipmentType.Sword, EquipmentSlot.Weapon)
                );
            case PvETarget.Uruk:
                return new Character(
                    CharacterType.Uruk,
                    new Stats(40, 7, 6, 4, 4, 4),
                    new Equipment(EquipmentType.Axe, EquipmentSlot.Weapon)
                );
            case PvETarget.Legolas:
                return new Character(
                    CharacterType.Legolas,
                    new Stats(45, 8, 7, 5, 5, 5),
                    new Equipment(EquipmentType.Bow, EquipmentSlot.Weapon)
                );
            case PvETarget.Batman:
                return new Character(
                    CharacterType.Batman,
                    new Stats(50, 9, 8, 6, 6, 6),
                    new Equipment(EquipmentType.Batarang, EquipmentSlot.Weapon)
                );
            default:
                throw new Error(`Unknown PvE target: ${target}`);
        }
    }
} 

File: backend/P2ERPG.Business/Managers/Battle/PvETargetManager.cs
using BilbolStack.Boonamai.P2ERPG.Business.Records.Characters;
using BilbolStack.Boonamai.P2ERPG.Common.Enumeration;
using System.Reflection.Metadata.Ecma335;

namespace BilbolStack.Boonamai.P2ERPG.Business.Managers.Battle
{
    public class PvETargetManager : IPvETargetManager
    {
        public Character GetTarget(PvETarget target)
        {
            return target switch
            {
                PvETarget.WoodenDummy => new Character(1, 1, "Wooden Dummy", CharacterType.Other, 10, 0, 0, 0, 0, 0, 0, 0),
                PvETarget.Chicken => new Character(1, 1, "Chicken", CharacterType.Other, 15, 2, 1, 0, 0, 0, 0, 0),
                PvETarget.Dogo => new Character(1, 1, "Dogo", CharacterType.Other, 20, 3, 2, 0, 0, 0, 0, 0),
                PvETarget.DaphneBlake => new Character(1, 1, "Daphne Blake", CharacterType.Human, 25, 4, 3, 1, 1, 1, 0, 0),
                PvETarget.Thug => new Character(1, 1, "Thug", CharacterType.Human, 30, 5, 4, 2, 2, 2, 0, 0),
                PvETarget.Sellsword => new Character(1, 1, "Sellsword", CharacterType.Human, 35, 6, 5, 3, 3, 3, 0, 0),
                PvETarget.Uruk => new Character(1, 1, "Uruk", CharacterType.Orc, 40, 7, 6, 4, 4, 4, 0, 0),
                PvETarget.Legolas => new Character(1, 1, "Legolas", CharacterType.Elf, 45, 8, 7, 5, 5, 5, 0, 0),
                PvETarget.Batman => new Character(1, 1, "Batman", CharacterType.Human, 50, 9, 8, 6, 6, 6, 0, 0),
                _ => throw new ArgumentException($"Unknown PvE target: {target}")
            };
        }

        public ArmorType GetArmorType(PvETarget target)
        {
            return target switch
            {
                PvETarget.WoodenDummy => ArmorType.BirthdaySuit,
                PvETarget.Chicken => ArmorType.BirthdaySuit,
                PvETarget.Dogo => ArmorType.BirthdaySuit,
                PvETarget.DaphneBlake => ArmorType.Light,
                PvETarget.Thug => ArmorType.Light,
                PvETarget.Sellsword => ArmorType.Medium,
                PvETarget.Uruk => ArmorType.Heavy,
                PvETarget.Legolas => ArmorType.Medium,
                PvETarget.Batman => ArmorType.Heavy,
                _ => throw new ArgumentException($"Unknown PvE target: {target}")
            };
        }

        public ShieldType GetShieldType(PvETarget target)
        {
            return target switch
            {
                PvETarget.WoodenDummy => ShieldType.None,
                PvETarget.Chicken => ShieldType.None,
                PvETarget.Dogo => ShieldType.None,
                PvETarget.DaphneBlake => ShieldType.Small,
                PvETarget.Thug => ShieldType.Medium,
                PvETarget.Sellsword => ShieldType.Medium,
                PvETarget.Uruk => ShieldType.None,
                PvETarget.Legolas => ShieldType.Medium,
                PvETarget.Batman => ShieldType.None,
                _ => throw new ArgumentException($"Unknown PvE target: {target}")
            };
        }

        public WeaponType GetWeaponType(PvETarget target)
        {
            return target switch
            {
                PvETarget.WoodenDummy => WeaponType.Fists,
                PvETarget.Chicken => WeaponType.Fists,
                PvETarget.Dogo => WeaponType.Fists,
                PvETarget.DaphneBlake => WeaponType.Fists,
                PvETarget.Thug => WeaponType.Mace,
                PvETarget.Sellsword => WeaponType.Sword,
                PvETarget.Uruk => WeaponType.TwoHandedMace,
                PvETarget.Legolas => WeaponType.Bow,
                PvETarget.Batman => WeaponType.TwoHandedSword,
                _ => throw new ArgumentException($"Unknown PvE target: {target}")
            };
        }
    }
} 